adapters/beachfront/beachfront.go

```diff
@@ -111,300 +113,334 @@ type BeachfrontResponseSlot struct {
 
-	if req.App != nil {
-		beachfrontReq.Domain = req.App.Domain
-		beachfrontReq.Page = req.App.ID
-	} else {
-		protoUrl := strings.Split(req.Site.Page, "//")
-		var domainPage string
-		// Resolves a panic for any Site.Page that does not include the protocol
-		if len(protoUrl) > 1 {
-			domainPage = protoUrl[1]
+	// The request that gets to here only has imps that contain a banner element. They may also contain
+	// a video element, but those are being ignored in this function.
+	for i := 0; i < len(request.Imp); i++ {
+
+		beachfrontExt, err := getBeachfrontExtension(request.Imp[i])
+		slot := beachfrontSlot{}
+
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		slot.Bidfloor = beachfrontExt.BidFloor
+		slot.Slot = request.Imp[i].ID
+
+		if beachfrontExt.AppId != "" {
+			slot.Id = beachfrontExt.AppId
 		} else {
-			domainPage = protoUrl[0]
+			slot.Id = beachfrontExt.AppIds.Banner
 		}
-		beachfrontReq.Domain = strings.Split(domainPage, "/")[0]
-		beachfrontReq.Page = req.Site.Page
-	}
 
-	beachfrontReq.RequestID = req.ID
+		for j := 0; j < len(request.Imp[i].Banner.Format); j++ {
 
-	return beachfrontReq, errs, imps
-}
+			slot.Sizes = append(slot.Sizes, beachfrontSize{
+				H: request.Imp[i].Banner.Format[j].H,
+				W: request.Imp[i].Banner.Format[j].W,
+			})
+		}
 
-func getVideoRequest(req *openrtb.BidRequest) (BeachfrontVideoRequest, []error, int) {
-	var videoImpsIndex = 0
-	var beachfrontReq = NewBeachfrontVideoRequest()
-	var errs = make([]error, 0, len(req.Imp))
-	var imps = 0
+		bfBannerRequest.Slots = append(bfBannerRequest.Slots, slot)
+	}
 
-	if req.App != nil {
-		if req.App.Domain != "" {
-			beachfrontReq.Site.Domain = req.App.Domain
-			beachfrontReq.Site.Page = req.App.ID
+	if request.Device != nil {
+		bfBannerRequest.IP = request.Device.IP
+		bfBannerRequest.DeviceModel = request.Device.Model
+		bfBannerRequest.DeviceOs = request.Device.OS
+		if request.Device.DNT != nil {
+			bfBannerRequest.Dnt = *request.Device.DNT
 		}
-	} else {
-		if req.Site.Page != "" {
-			if req.Site.Domain == "" {
-				if strings.Contains(req.Site.Page, "//") {
-					// Remove protocol if exists
-					beachfrontReq.Site.Domain = strings.Split(req.Site.Page, "//")[1]
-				}
-				if strings.Contains(beachfrontReq.Site.Domain, "/") {
-					// Drop everything after the first "/"
-					beachfrontReq.Site.Domain = strings.Split(beachfrontReq.Site.Domain, "/")[0]
-				}
-			} else {
-				beachfrontReq.Site.Domain = req.Site.Domain
-			}
-			beachfrontReq.Site.Page = req.Site.Page
+		if request.Device.UA != "" {
+			bfBannerRequest.UA = request.Device.UA
 		}
 	}
 
-	/*
-		The req could contain banner,audio,native and video imps when It arrives here. I am only
-		interested in video
+	site := getSite(request)
+	bfBannerRequest.IsMobile = site.Mobile
+	bfBannerRequest.Page = site.Page
+	bfBannerRequest.Domain = site.Domain
+
+	if request.User != nil {
+		if bfBannerRequest.User.ID == "" {
```

Comment:
Shouldn't this `if` condition rather be: `if request.User.ID != ""` and similar for the next `if` condition too?

Reply:
Yes. I got in a hurry and made a dumb mistake.